mbuild/bond_graph.py

```diff
@@ -188,3 +58,7 @@ def _bfs(self, source):
                     yield v
                     seen.add(v)
                     nextlevel.update(self.neighbors(v))
+
+    def connected_components(self):
+        """Return list of connected bond component of bondgraph."""
+        return [list(mol) for mol in nx.connected_components(self)]
```

Comment:
Change to gmso from_mbuild _parse_molecule_residue function (#<----changed line here) denotes a modification.
```python
def _parse_molecule_residue(site_map, compound):
     """Parse information necessary for residue and molecule labels when converting from mbuild."""
     connected_subgraph = compound.bond_graph.connected_components()
     molecule_tracker = dict()
     residue_tracker = dict()
     for molecule in connected_subgraph:
         if len(molecule) == 1:
             ancestors = [molecule.pop()] #<----changed line here
         else:
             ancestors = IndexedSet(molecule.pop().ancestors()) #<----changed line here
             for particle in molecule:
                 # This works because particle.ancestors traversed, and hence
                 # the lower level will be in the front.
                 # The intersection will be left at the end,
                 # ancestor of the first particle is used as reference.
                 # Hence, this called will return the lowest-level Compound]
                 # that is a molecule
                 ancestors = ancestors.intersection(
                     IndexedSet(particle.ancestors())
                 )
 
         """Parse molecule information"""
         molecule_tag = ancestors[0]
         # The duplication is determined solely by name
         if molecule_tag.name in molecule_tracker:
             molecule_tracker[molecule_tag.name] += 1
         else:
             molecule_tracker[molecule_tag.name] = 0
         molecule_number = molecule_tracker[molecule_tag.name]
         """End of molecule parsing"""

     """Start of residue parsing""" #<----changed line here
     connected_subgraph = compound.bond_graph.connected_components() #<----changed line here
     for molecule in connected_subgraph: #<----changed line here
         for particle in molecule:
             """Parse residue information"""
             residue_tag = (
                 particle if not particle.n_direct_bonds else particle.parent
             )
             if residue_tag.name in residue_tracker:
                 if residue_tag not in residue_tracker[residue_tag.name]:
                     residue_tracker[residue_tag.name][residue_tag] = len(
                         residue_tracker[residue_tag.name]
                     )
             else:
                 residue_tracker[residue_tag.name] = {residue_tag: 0}
 
             residue_number = residue_tracker[residue_tag.name][residue_tag]
             site_map[particle]["residue"] = (residue_tag.name, residue_number)
             site_map[particle]["molecule"] = (
                 molecule_tag.name,
                 molecule_number,
             )

Reply:
Probably good to do it in a separate PR.